src/jit/assertionprop.cpp

```diff
@@ -68,8 +65,10 @@ void Compiler::optAddCopies()
         return;
     }
 
-    for (lclNum = 0, varDsc = lvaTable; lclNum < lvaCount; lclNum++, varDsc++)
+    for (unsigned lclNum = 0; lclNum < lvaCount; lclNum++)
     {
+        LclVarDsc* varDsc = &lvaTable[lclNum];
```

Comment:
That's a fair point.

One other change I had considered was to add `LclVarDsc* operator+(unsigned index)` to `LclVarTable`, which would avoid the need to update most of these sites in any case.

I think I'd prefer to keep things as-is, though, as I find references generally preferable to pointers given the nullability issue. My intent is to gradually replace the various uses of `LclVarDsc*` with `LclVarDsc&`, so this should eventually be a non-issue.